### PR TITLE
use cached connection in boto_* modules

### PR DESCRIPTION
**Pull request by mgwilliams**
@s0undt3ch -- I had to totally disable F821 in the boto_\* modules because of the `setattr()` on the module to get `_get_conn` and `_cache_id` into the global namespace. Would it be better to add these vars to .testing.pylintrc?

### CODE REVIEW

**Comment by mgwilliams** [comment]
FYI @ryan-lane, @claudiupopescu, @tsaridas

**Comment by mgwilliams** [comment]
@s0undt3ch could you provide any insight into why my `#pylint disable=E0602` is not working?

File: salt/modules/boto_asg.py [hunk 1]
```diff
@@ -33,11 +33,14 @@
 
 :depends: boto
 '''
+# keep lint from choking on _get_conn and _cache_id
+#pylint disable=E0602
```
Comment (s0undt3ch) [inline salt/modules/boto_asg.py]:
```
# pylint: disable=E0602
```

Notice the missing `:`

**Comment by s0undt3ch** [comment]
Since this is boto specific and not salt specific, I'd prefer not to have this in the global lint file.

Also, I haven't fully read your changes but `__context__` does not work to cache your connection object?

I think I might need to revisit that code to check...

**Comment by rallytime** [comment]
@mgwilliams Looks like this needs a rebase.

**Comment by mgwilliams** [comment]
@s0undt3ch: I am using `__context__` to cache the connction. Since this is highly repetitive code, though, it has been moved into `salt.utils.boto`. This PR, for the most part, is just updating all the modules to use the utils function.

@rallytime: hopefully these latest commits will fix the conflict and the lint errors.

**Comment by s0undt3ch** [comment]
OK. Sorry I'm on my phone which makes it harder to review the whole PR code.

**Comment by mgwilliams** [comment]
@s0undt3ch what does this mean in the lint failure log?

```21:11:15 Commits don't match 46bde83216dbf87cde4f4b366ed8b55681a5065d == 8fa76d7e40a2d2b1d705eac24ceb8d342e9692cb```

**Comment by s0undt3ch** [comment]
My first guess is that you rebased and force pushed to the branch which might have triggered 2 builds with different sha's. Let me stop this one and re-trigger.

**Comment by mgwilliams** [comment]
@s0undt3ch I merged develop in (and fixed some conflicts), but did not force push.

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by s0undt3ch** [comment]
That's weird. The cloned repo didn't match the sha being tested. Let's see how this run goes...

**Pull request merged**